erroneous error: 'the overlay template '/etc/hosts.ww' does not exists in 'host' overlay

I just encountered this running `development` on an el9 system. I don't think that el9 is relevant here; but just in case.

```bash
[demo@demo2 warewulf]$ sudo wwctl configure hostfile
ERROR  : 'the overlay template '/etc/hosts.ww' does not exists in 'host' overlay

[demo@demo2 warewulf]$ sudo wwctl overlay list host -a
OVERLAY NAME                   FILES/DIRS  
host                           /etc/        
host                           /etc/dhcp/   
host                           /etc/dhcp/dhcpd.conf.ww
host                           /etc/dhcpd.conf
host                           /etc/exports.ww
host                           /etc/hosts.ww
host                           /etc/profile.d/
host                           /etc/profile.d/ssh_setup.csh
host                           /etc/profile.d/ssh_setup.sh
```

@mslacken I think I've fixed _this_ problem with the following patch:

```
diff --git a/internal/pkg/configure/hostfile.go b/internal/pkg/configure/hostfile.go
index 15e0b7a..bfb6bb2 100644
--- a/internal/pkg/configure/hostfile.go
+++ b/internal/pkg/configure/hostfile.go
@@ -15,7 +15,9 @@ import (
 Creates '/etc/hosts' from the host template.
 */
 func Hostfile() error {
-       if !(util.IsFile(path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"))) {
+       hostTemplate := path.Join(overlay.OverlaySourceDir("host"), "/etc/hosts.ww")
+       if !(util.IsFile(hostTemplate)) {
                wwlog.Error("'the overlay template '/etc/hosts.ww' does not exists in 'host' overlay")
                os.Exit(1)
        }
@@ -24,13 +26,13 @@ func Hostfile() error {
        hostname, _ := os.Hostname()
        nodeInfo.Id.Set(hostname)
        buffer, backupFile, writeFile, err := overlay.RenderTemplateFile(
-               path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"),
+               hostTemplate,
                tstruct)
        if err != nil {
                wwlog.Error("%s", err)
                os.Exit(1)
        }
-       info, err := os.Stat(path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"))
+       info, err := os.Stat(hostTemplate)
        if err != nil {
                wwlog.Error("%s", err)
                os.Exit(1)
```

But this has exposed another bug:

```
[demo@demo1 warewulf]$ sudo wwctl configure hostfile
panic: reflect: call of reflect.Value.Field on zero Value

goroutine 1 [running]:
reflect.Value.Field({0x0, 0x0, 0x0}, 0xbf2b22)
	/usr/lib/golang/src/reflect/value.go:1198 +0xef
github.com/hpcng/warewulf/internal/pkg/node.(*NodeConf).getterFrom(_, {{{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0x0, 0x0}, {0x0, ...}}, ...}, ...)
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/internal/pkg/node/transformers.go:93 +0x21e7
github.com/hpcng/warewulf/internal/pkg/node.(*NodeConf).GetFrom(...)
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/internal/pkg/node/transformers.go:27
github.com/hpcng/warewulf/internal/pkg/overlay.InitStruct({{{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0, ...}}, ...})
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/internal/pkg/overlay/datastructure.go:87 +0x992
github.com/hpcng/warewulf/internal/pkg/configure.Hostfile()
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/internal/pkg/configure/hostfile.go:25 +0x136
github.com/hpcng/warewulf/internal/app/wwctl/configure/hostfile.CobraRunE(0x1494a60, {0xd9c59f, 0x0, 0x0})
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/internal/app/wwctl/configure/hostfile/main.go:9 +0x17
github.com/spf13/cobra.(*Command).execute(0x1494a60, {0x1549180, 0x0, 0x0})
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:850 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x1494280)
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:958 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:895
main.main()
	/home/demo/rpmbuild/BUILD/warewulf-4.3.0/cmd/wwctl/main.go:13 +0x25
```

I think the issue is that we're passing an empty NodeInfo to overlay.InitStruct.

```
        var nodeInfo node.NodeInfo
        tstruct := overlay.InitStruct(nodeInfo)
```

But  I don't know the right way to initialize one in this instance. The only other call to InitStruct is in internal/app/wwctl/overlay/show/main.go.

```
		filteredNodes := node.FilterByName(nodes, []string{NodeName})
		[...]
		tstruct := overlay.InitStruct(filteredNodes[0])
```

@mslacken Pull request submitted.

`overlay.InitStruct` expects a `NodeInfo`; but `NewConf` produces a `NodeConf`; so I can't use it to initialize `nodeInfo`. If I do, I get a compile error.

```
../../internal/pkg/configure/hostfile.go:24:31: cannot use nodeInfo (type node.NodeConf) as type node.NodeInfo in argument to overlay.InitStruct
../../internal/pkg/configure/hostfile.go:26:10: nodeInfo.Id undefined (type node.NodeConf has no field or method Id)
```

--- a/internal/pkg/configure/hostfile.go
+++ b/internal/pkg/configure/hostfile.go
@@ -15,7 +15,8 @@ import (
 Creates '/etc/hosts' from the host template.
 */
 func Hostfile() error {
-	if !(util.IsFile(path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"))) {
+	hostTemplate := path.Join(overlay.OverlaySourceDir("host"), "/etc/hosts.ww")
+	if !(util.IsFile(hostTemplate)) {
 		wwlog.Error("'the overlay template '/etc/hosts.ww' does not exists in 'host' overlay")
 		os.Exit(1)
 	}
@@ -24,13 +25,13 @@ func Hostfile() error {
 	hostname, _ := os.Hostname()
 	nodeInfo.Id.Set(hostname)
 	buffer, backupFile, writeFile, err := overlay.RenderTemplateFile(
-		path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"),
+		hostTemplate,
 		tstruct)
 	if err != nil {
 		wwlog.Error("%s", err)
 		os.Exit(1)
 	}
-	info, err := os.Stat(path.Join(overlay.OverlaySourceDir("host"), "/host/etc/hosts.ww"))
+	info, err := os.Stat(hostTemplate)
 	if err != nil {
 		wwlog.Error("%s", err)
 		os.Exit(1)

--- a/internal/pkg/configure/hostfile.go
+++ b/internal/pkg/configure/hostfile.go
@@ -20,10 +20,11 @@ func Hostfile() error {
 		wwlog.Error("'the overlay template '/etc/hosts.ww' does not exists in 'host' overlay")
 		os.Exit(1)
 	}
-	var nodeInfo node.NodeInfo
-	tstruct := overlay.InitStruct(nodeInfo)
+
+	nodeInfo := node.NewInfo()
 	hostname, _ := os.Hostname()
 	nodeInfo.Id.Set(hostname)
+	tstruct := overlay.InitStruct(nodeInfo)
 	buffer, backupFile, writeFile, err := overlay.RenderTemplateFile(
 		hostTemplate,
 		tstruct)

--- a/internal/pkg/overlay/overlay.go
+++ b/internal/pkg/overlay/overlay.go
@@ -81,14 +81,11 @@ func BuildSpecificOverlays(nodes []node.NodeInfo, overlayNames []string) error {
 Build overlay for the host, so no argument needs to be given
 */
 func BuildHostOverlay() error {
-	var host node.NodeInfo
-	host.Kernel = new(node.KernelEntry)
-	host.Ipmi = new(node.IpmiEntry)
-	var idEntry node.Entry
+	host := node.NewInfo()
 	hostname, _ := os.Hostname()
+	host.Id.Set(hostname)
+
 	wwlog.Info("Building overlay for %s: host", hostname)
-	idEntry.Set(hostname)
-	host.Id = idEntry
 	hostdir := OverlaySourceDir("host")
 	stats, err := os.Stat(hostdir)
 	if err != nil {

